src/util/play_time.rs

```diff
@@ -0,0 +1,165 @@
+use crate::config::get_persistent_dir;
+use constants::filename::persistent;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    fs,
+    time::{Duration, Instant, SystemTime, UNIX_EPOCH},
+};
+
+mod serial_secs {
+    use super::*;
+
+    pub fn default_time() -> SystemTime {
+        SystemTime::now()
+    }
+
+    pub fn serialize<S>(t: &SystemTime, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let ts = t.duration_since(UNIX_EPOCH).unwrap_or_default().as_secs();
+        s.serialize_u64(ts)
+    }
+
+    pub fn deserialize<'de, D>(d: D) -> Result<SystemTime, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let secs = u64::deserialize(d)?;
+        Ok(UNIX_EPOCH + Duration::from_secs(secs))
+    }
+}
+
+mod serial_dur {
+    use super::*;
+    // Same logic, just distinct for Duration vs SystemTime types if needed
+    pub fn serialize<S>(d: &Duration, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        s.serialize_u64(d.as_secs())
+    }
+
+    pub fn deserialize<'de, D>(d: D) -> Result<Duration, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let secs = u64::deserialize(d)?;
+        Ok(Duration::from_secs(secs))
+    }
+}
+
+// REMOVED: Copy trait (SystemTime is not Copy)
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PlayTimeState {
```

Comment:
Consider renaming this to `PlayTimeTracker` or such, especially if you follow my suggestion to move the actual logic here